server/build.gradle

```diff
@@ -725,23 +723,40 @@ task war(type: Jar) {
   dependsOn rootProject.generateLicenseReport
   dependsOn configureWar
   dependsOn ':server:jar', ':server:compileAssets'
-  dependsOn project.configurations.compile
-  dependsOn project.configurations.api
+  dependsOn project.configurations.compileClasspath
+  dependsOn project.configurations.allApis
   dependsOn project.configurations.spark
   finalizedBy 'verifyWar'
   archiveBaseName = 'cruise'
-  extension 'war'
+  archiveExtension = 'war'
+
+  // Using a Set<String> is more reliable than Set<File> because File.equals() may be affected by other internal fields
+  // aside from `path` that aren't important to us; thus, Set<File> may allow duplicate paths and Set<String> will not.
+  Set<String> classpath = (
```

Comment:
In other words, `Set<String>` is better at deduplicating paths to jars than `Set<File>`.